src/Tests/FlatSharpCompilerTests/UnionTests.cs

```diff
@@ -109,5 +109,20 @@ struct ValueStruct (fs_valueStruct) {{ x : int; }}
             new()));
 
         Assert.Contains("FlatSharp unions may not contain duplicate types. Union = UnionTests.MyUnion", ex.Message);
+    }
+
+    [Fact]
+    public void Union_WithCustomEnumValue()
+    {
+        string schema = @"
+            namespace UnionTests;
+            table A {}
+            table B {}
+            union C { A = 2, B = 4 }
```

Comment:
I try not to use this method of testing any longer where I just ensure it compiles. A better test here would show the impact that `A = 2` and `B = 4` will have. A better project for that kind of thing is probably `FlatSharpEndToEndTests`. This test project is mostly for metadata and error cases.